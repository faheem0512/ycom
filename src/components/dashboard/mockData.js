export const successData = {
    "hits": [
        {
            "created_at": "2015-04-29T16:20:57.000Z",
            "title": "Microsoft Launches Visual Studio Code, a Free Cross-Platform Code Editor",
            "url": "http://techcrunch.com/2015/04/29/microsoft-shocks-the-world-with-visual-studio-code-a-free-code-editor-for-os-x-linux-and-windows/",
            "author": "MikusR",
            "points": 1608,
            "story_text": null,
            "comment_text": null,
            "num_comments": 542,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1430324457,
            "relevancy_score": 5995,
            "_tags": [
                "story",
                "author_MikusR",
                "story_9459364"
            ],
            "objectID": "9459364",
            "_highlightResult": {
                "title": {
                    "value": "Microsoft Launches Visual Studio Code, a Free Cross-Platform Code Editor",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "http://techcrunch.com/2015/04/29/microsoft-shocks-the-world-with-visual-studio-code-a-free-code-editor-for-os-x-linux-and-windows/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "MikusR",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2019-09-18T10:11:18.000Z",
            "title": "Colorado Town Offers 1 Gbps for $60 After Years of Battling Comcast",
            "url": "https://www.techdirt.com/articles/20190904/08392642916/colorado-town-offers-1-gbps-60-after-years-battling-comcast.shtml",
            "author": "CrankyBear",
            "points": 1606,
            "story_text": null,
            "comment_text": null,
            "num_comments": 498,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1568801478,
            "_tags": [
                "story",
                "author_CrankyBear",
                "story_21003773"
            ],
            "objectID": "21003773",
            "_highlightResult": {
                "title": {
                    "value": "Colorado Town Offers 1 Gbps for $60 After Years of Battling Comcast",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://www.techdirt.com/articles/20190904/08392642916/colorado-town-offers-1-gbps-60-after-years-battling-comcast.shtml",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "CrankyBear",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2010-07-24T23:07:27.000Z",
            "title": null,
            "url": null,
            "author": "Ardit20",
            "points": 1606,
            "story_text": "Yes",
            "comment_text": null,
            "num_comments": null,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": 1544581,
            "created_at_i": 1280012847,
            "relevancy_score": 2649,
            "_tags": [
                "pollopt",
                "author_Ardit20",
                "story_1544582"
            ],
            "objectID": "1544582",
            "_highlightResult": {
                "author": {
                    "value": "Ardit20",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "Yes",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2018-01-05T17:29:06.000Z",
            "title": "Why Raspberry Pi Isn't Vulnerable to Spectre or Meltdown",
            "url": "https://www.raspberrypi.org/blog/why-raspberry-pi-isnt-vulnerable-to-spectre-or-meltdown/",
            "author": "MikusR",
            "points": 1603,
            "story_text": null,
            "comment_text": null,
            "num_comments": 215,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1515173346,
            "relevancy_score": 7877,
            "_tags": [
                "story",
                "author_MikusR",
                "story_16080002"
            ],
            "objectID": "16080002",
            "_highlightResult": {
                "title": {
                    "value": "Why Raspberry Pi Isn't Vulnerable to Spectre or Meltdown",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://www.raspberrypi.org/blog/why-raspberry-pi-isnt-vulnerable-to-spectre-or-meltdown/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "MikusR",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2020-03-27T14:18:52.000Z",
            "title": "Zoom needs to clean up its privacy act",
            "url": "https://blogs.harvard.edu/doc/2020/03/27/zoom/",
            "author": "seapunk",
            "points": 1597,
            "story_text": null,
            "comment_text": null,
            "num_comments": 509,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1585318732,
            "_tags": [
                "story",
                "author_seapunk",
                "story_22703000"
            ],
            "objectID": "22703000",
            "_highlightResult": {
                "title": {
                    "value": "Zoom needs to clean up its privacy act",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://blogs.harvard.edu/doc/2020/03/27/zoom/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "seapunk",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2019-11-28T09:59:47.000Z",
            "title": "Firefox Replay",
            "url": "https://firefox-replay.com/",
            "author": "nachtigall",
            "points": 1596,
            "story_text": null,
            "comment_text": null,
            "num_comments": 285,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1574935187,
            "_tags": [
                "story",
                "author_nachtigall",
                "story_21655958"
            ],
            "objectID": "21655958",
            "_highlightResult": {
                "title": {
                    "value": "Firefox Replay",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://firefox-replay.com/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "nachtigall",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2019-10-22T14:02:14.000Z",
            "title": "Firefox 70",
            "url": "https://hacks.mozilla.org/2019/10/firefox-70-a-bountiful-release-for-all/",
            "author": "feross",
            "points": 1595,
            "story_text": null,
            "comment_text": null,
            "num_comments": 457,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1571752934,
            "_tags": [
                "story",
                "author_feross",
                "story_21323663"
            ],
            "objectID": "21323663",
            "_highlightResult": {
                "title": {
                    "value": "Firefox 70",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://hacks.mozilla.org/2019/10/firefox-70-a-bountiful-release-for-all/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "feross",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2011-10-24T19:44:59.000Z",
            "title": "John McCarthy Has Died",
            "url": "http://en.wikipedia.org/wiki/John_McCarthy_(computer_scientist)",
            "author": "apgwoz",
            "points": 1595,
            "story_text": "",
            "comment_text": null,
            "num_comments": 218,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1319485499,
            "relevancy_score": 3536,
            "_tags": [
                "story",
                "author_apgwoz",
                "story_3151233"
            ],
            "objectID": "3151233",
            "_highlightResult": {
                "title": {
                    "value": "John McCarthy Has Died",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "http://en.wikipedia.org/wiki/John_McCarthy_(computer_scientist)",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "apgwoz",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2020-02-05T15:09:22.000Z",
            "title": "Wacom tablets track every app you open",
            "url": "https://robertheaton.com/2020/02/05/wacom-drawing-tablets-track-name-of-every-application-you-open/",
            "author": "krithix",
            "points": 1594,
            "story_text": null,
            "comment_text": null,
            "num_comments": 449,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1580915362,
            "_tags": [
                "story",
                "author_krithix",
                "story_22247292"
            ],
            "objectID": "22247292",
            "_highlightResult": {
                "title": {
                    "value": "Wacom tablets track every app you open",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://robertheaton.com/2020/02/05/wacom-drawing-tablets-track-name-of-every-application-you-open/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "krithix",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2013-09-11T19:47:51.000Z",
            "title": null,
            "url": null,
            "author": "sinak",
            "points": 1593,
            "story_text": "I want more NSA news. NSA stories should be allowed to rank naturally, with less filtering/flagging.",
            "comment_text": null,
            "num_comments": null,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": 6369530,
            "created_at_i": 1378928871,
            "relevancy_score": 4853,
            "_tags": [
                "pollopt",
                "author_sinak",
                "story_6369532"
            ],
            "objectID": "6369532",
            "_highlightResult": {
                "author": {
                    "value": "sinak",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "I want more NSA news. NSA stories should be allowed to rank naturally, with less filtering/flagging.",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2011-07-29T14:47:55.000Z",
            "title": null,
            "url": null,
            "author": "ColinWright",
            "points": 1585,
            "story_text": "Provide a \"merge\" mechanism",
            "comment_text": null,
            "num_comments": null,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": 2822041,
            "created_at_i": 1311950875,
            "relevancy_score": 3361,
            "_tags": [
                "pollopt",
                "author_ColinWright",
                "story_2822045"
            ],
            "objectID": "2822045",
            "_highlightResult": {
                "author": {
                    "value": "ColinWright",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "Provide a \"merge\" mechanism",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2015-02-27T17:12:42.000Z",
            "title": "Leonard Nimoy, Spock of ‘Star Trek,’ Dies at 83",
            "url": "http://www.nytimes.com/2015/02/27/arts/television/leonard-nimoy-spock-of-star-trek-dies-at-83.html?smid=tw-bna",
            "author": "coloneltcb",
            "points": 1584,
            "story_text": null,
            "comment_text": null,
            "num_comments": 165,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1425057162,
            "relevancy_score": 5874,
            "_tags": [
                "story",
                "author_coloneltcb",
                "story_9119951"
            ],
            "objectID": "9119951",
            "_highlightResult": {
                "title": {
                    "value": "Leonard Nimoy, Spock of ‘Star Trek,’ Dies at 83",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "http://www.nytimes.com/2015/02/27/arts/television/leonard-nimoy-spock-of-star-trek-dies-at-83.html?smid=tw-bna",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "coloneltcb",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2014-02-10T15:26:52.000Z",
            "title": "Facebook Fraud [video]",
            "url": "http://www.youtube.com/watch?v=oVfHeWTKjag",
            "author": "fanfantm",
            "points": 1583,
            "story_text": "",
            "comment_text": null,
            "num_comments": 275,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1392046012,
            "relevancy_score": 5149,
            "_tags": [
                "story",
                "author_fanfantm",
                "story_7211514"
            ],
            "objectID": "7211514",
            "_highlightResult": {
                "title": {
                    "value": "Facebook Fraud [video]",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "http://www.youtube.com/watch?v=oVfHeWTKjag",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "fanfantm",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2020-02-04T17:30:40.000Z",
            "title": "Why Discord is switching from Go to Rust",
            "url": "https://blog.discordapp.com/why-discord-is-switching-from-go-to-rust-a190bbca2b1f",
            "author": "Sikul",
            "points": 1582,
            "story_text": null,
            "comment_text": null,
            "num_comments": 640,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1580837440,
            "_tags": [
                "story",
                "author_Sikul",
                "story_22238335"
            ],
            "objectID": "22238335",
            "_highlightResult": {
                "title": {
                    "value": "Why Discord is switching from Go to Rust",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://blog.discordapp.com/why-discord-is-switching-from-go-to-rust-a190bbca2b1f",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "Sikul",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2016-05-11T06:50:52.000Z",
            "title": "Introducing unlimited private repositories",
            "url": "https://github.com/blog/2164-introducing-unlimited-private-repositories",
            "author": "fuzionmonkey",
            "points": 1576,
            "story_text": null,
            "comment_text": null,
            "num_comments": 631,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1462949452,
            "relevancy_score": 6721,
            "_tags": [
                "story",
                "author_fuzionmonkey",
                "story_11673103"
            ],
            "objectID": "11673103",
            "_highlightResult": {
                "title": {
                    "value": "Introducing unlimited private repositories",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://github.com/blog/2164-introducing-unlimited-private-repositories",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "fuzionmonkey",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2019-11-13T23:17:23.000Z",
            "title": "John Carmack: I’m going to work on artificial general intelligence",
            "url": "https://www.facebook.com/100006735798590/posts/2547632585471243/",
            "author": "jbredeche",
            "points": 1574,
            "story_text": null,
            "comment_text": null,
            "num_comments": 887,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1573687043,
            "_tags": [
                "story",
                "author_jbredeche",
                "story_21530860"
            ],
            "objectID": "21530860",
            "_highlightResult": {
                "title": {
                    "value": "John Carmack: I’m going to work on artificial general intelligence",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://www.facebook.com/100006735798590/posts/2547632585471243/",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "jbredeche",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2016-03-22T22:40:51.000Z",
            "title": "I've Just Liberated My Modules",
            "url": "https://medium.com/@azerbike/i-ve-just-liberated-my-modules-9045c06be67c",
            "author": "chejazi",
            "points": 1573,
            "story_text": null,
            "comment_text": null,
            "num_comments": 495,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1458686451,
            "relevancy_score": 6627,
            "_tags": [
                "story",
                "author_chejazi",
                "story_11340510"
            ],
            "objectID": "11340510",
            "_highlightResult": {
                "title": {
                    "value": "I've Just Liberated My Modules",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://medium.com/@azerbike/i-ve-just-liberated-my-modules-9045c06be67c",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "chejazi",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2010-07-04T23:02:36.000Z",
            "title": null,
            "url": null,
            "author": "cmelbye",
            "points": 1573,
            "story_text": "Nonreligious/Atheist/Agnostic/etc",
            "comment_text": null,
            "num_comments": null,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": 1486594,
            "created_at_i": 1278284556,
            "relevancy_score": 2608,
            "_tags": [
                "pollopt",
                "author_cmelbye",
                "story_1486597"
            ],
            "objectID": "1486597",
            "_highlightResult": {
                "author": {
                    "value": "cmelbye",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "Nonreligious/Atheist/Agnostic/etc",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2018-09-06T17:51:10.000Z",
            "title": "Chrome 69: “www.” subdomain missing from URL",
            "url": "https://bugs.chromium.org/p/chromium/issues/detail?id=881410",
            "author": "gouggoug",
            "points": 1572,
            "story_text": null,
            "comment_text": null,
            "num_comments": 876,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": null,
            "created_at_i": 1536256270,
            "relevancy_score": 8347,
            "_tags": [
                "story",
                "author_gouggoug",
                "story_17927972"
            ],
            "objectID": "17927972",
            "_highlightResult": {
                "title": {
                    "value": "Chrome 69: “www.” subdomain missing from URL",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "https://bugs.chromium.org/p/chromium/issues/detail?id=881410",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "gouggoug",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        },
        {
            "created_at": "2011-12-01T08:57:22.000Z",
            "title": "",
            "url": "",
            "author": "sasvari",
            "points": 1572,
            "story_text": "United States - NYC and SF area",
            "comment_text": null,
            "num_comments": null,
            "story_id": null,
            "story_title": null,
            "story_url": null,
            "parent_id": 3298905,
            "created_at_i": 1322729842,
            "relevancy_score": 3603,
            "_tags": [
                "pollopt",
                "author_sasvari",
                "story_3298906"
            ],
            "objectID": "3298906",
            "_highlightResult": {
                "title": {
                    "value": "",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "url": {
                    "value": "",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "author": {
                    "value": "sasvari",
                    "matchLevel": "none",
                    "matchedWords": []
                },
                "story_text": {
                    "value": "United States - NYC and SF area",
                    "matchLevel": "none",
                    "matchedWords": []
                }
            }
        }
    ],
    "nbHits": 21725285,
    "page": 12,
    "nbPages": 50,
    "hitsPerPage": 20,
    "exhaustiveNbHits": true,
    "query": "",
    "params": "advancedSyntax=true&analytics=true&analyticsTags=backend&page=12",
    "processingTimeMS": 1
};