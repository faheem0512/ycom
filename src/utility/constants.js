export const HEADER_VALUES = ["Comments", "Vote Count", "Up Vote", "News Details"];