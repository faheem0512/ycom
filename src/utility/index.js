import moment from 'moment';

export function getDomainName(url) {
    return new URL(url).hostname;
}

export function getRelativeTime(timestamp) {
    return moment(timestamp).fromNow()
}

export function setInStorage(key,value) {
    try {
        window.localStorage.setItem(key, JSON.stringify(value));
    } catch (e) {
        console.log(e);  // eslint-disable-line
    }
}

export function getFromStorage(key){
    try {
        const item = window.localStorage.getItem(key);
        return JSON.parse(item);
    } catch (e) {
        console.log(e); // eslint-disable-line
    }
}
