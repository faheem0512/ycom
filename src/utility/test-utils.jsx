import React from "react";
import { render } from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "../redux/configureStore";
import initialState from "../redux/initialState";
import { Router,Route } from "react-router-dom";
import { createMemoryHistory } from "history";

const store = configureStore(initialState);

const RouterMatchProvider = ({ children }) => {
  return <Provider store={store}>
    <Router history={createMemoryHistory({ initialEntries: ['/'] })}>
      <Route path={'/'} component={children} />
    </Router>
  </Provider>;
};


const renderWithRouterMatch = (ui,...options) => {
  return render(ui,{wrapper:RouterMatchProvider, ...options });
};



const customRender = (ui, ...options) => {
  return render(<Provider store={store}>
    {ui}
  </Provider>,{ ...options });
};

export * from "@testing-library/react";
export { customRender, renderWithRouterMatch };
