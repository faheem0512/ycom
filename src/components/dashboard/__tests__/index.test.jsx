import React from "react";
import { renderWithRouterMatch } from "../../../utility/test-utils";
import Dashboard from "../index";
import {successData} from "../mockData";




describe('Dashboard',()=>{

    it("render  dashboard", () => {
        const { container  } = renderWithRouterMatch(Dashboard);
        process.nextTick(() => expect(container).toMatchSnapshot());
    });

    it("fetchData with success", () => {
        const { getAllByTestId  } = renderWithRouterMatch(Dashboard);
        let localMockFetch = mockFetch(successData);
        expect(fetch).toHaveBeenCalledTimes(0);
        process.nextTick(() => {
            // expect(getAllByTestId("dashboard")[1]).toHaveTextContent(/Microsoft Launches/);
            localMockFetch.mockClear();
        });
    });


});


const mockFetch = (data, isReject) => {
    /*
        There is a known compatibility issue with React DOM 16.8 where you will see the following warning:
        Warning: An update to ComponentName inside a test was not wrapped in act(...)
      * */
    // @ts-ignore
    return jest.spyOn(window, "fetch").mockImplementation(() => {
        if (isReject) {
            return Promise.resolve({
                json: () => Promise.reject(data),
                statusText: "Error"
            });
        }
        return Promise.resolve({
            json: () => Promise.resolve(data),
            ok: true
        });
    });
};
