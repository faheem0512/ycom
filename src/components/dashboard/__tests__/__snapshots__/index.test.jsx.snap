// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render  dashboard 1`] = `
<div>
  <div
    data-testid="dashboard"
  >
    <div
      class="loader-backdrop"
    >
      <div
        id="loader"
        class="background"
      >
        <div
          class="dots container"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
